cpp.mk

```diff
@@ -44,9 +44,8 @@ $1_DEPS := $$(patsubst %.cpp,$${BUILD_DIR}/$1/%.d,$${$1_SOURCES})
 # header files)
 $1_OBJECT_DIR := $${BUILD_DIR}/$1
 
-# Depend on the depends file so we'll get rebuilt if it's missing
-# (which will have the side effect of re-producing the depends file)
-$${$1_OBJS} : $${$1_OBJECT_DIR}/%.o : %.cpp $${BUILD_DIR}/$1/%.d
+# Object files are produced by compiling source files
+$${$1_OBJS} : $${$1_OBJECT_DIR}/%.o : %.cpp
```

Comment:
Wait no, we want to touch the `.o` file, not the `.d`.  So that model solution looks as though it's going to hit exactly the issue I'm trying to avoid.

I still favour the original fix to be honest... let's talk again.